### DISCUSSION TITLE
Is react-email production ready?

### DISCUSSION THREAD

**Discussion by zanzlender** [resend/react-email#857] [General]
I've recently stumbled across this library and found it to be revolutionary! However I soon realized some problems, the biggest one being not properly applying styles even with simple styling...

This is how my email is supposed to look and looks in dev.

![image](https://github.com/resendlabs/react-email/assets/44570474/ed451a34-e10e-488e-9c21-f00cc7fdc492)

In the next few chapters I will post how it looks once I open it in my 2 providers I use most often, Gmail and Outlook.

### Outlook

![image](https://github.com/resendlabs/react-email/assets/44570474/0a6b076a-873a-4c38-a1d8-fd1c8309ad41)

### Gmail

![image](https://github.com/resendlabs/react-email/assets/44570474/362bc7f7-6b99-4a9f-a016-e65e55c5206c)

Here I thought, hey maybe Tailwind is the problem so I wanted to recreate it using plain css, and here is the result.

### Outlook

![image](https://github.com/resendlabs/react-email/assets/44570474/59f3399a-fec6-4ec0-aae4-ca7d2a83ea75)

### Gmail

![image](https://github.com/resendlabs/react-email/assets/44570474/3701cff2-13b8-467b-bb00-f0b71ac9fc37)

Now I know that this is still in betaish but still this seems extremely weird... Or am I doing something wrong? Like am I not supposed to use flex or such things?

**Comment by bukinoshita** [comment]
> I've recently stumbled across this library and found it to be revolutionary! However I soon realized some problems, the biggest one being not properly applying styles even with simple styling...
> 
> This is how my email is supposed to look and looks in dev.
> 
> ![image](https://user-images.githubusercontent.com/44570474/255142034-ed451a34-e10e-488e-9c21-f00cc7fdc492.png)
> 
> In the next few chapters I will post how it looks once I open it in my 2 providers I use most often, Gmail and Outlook.
> 
> ## With TailwindCSS
> ### Outlook
> ![image](https://user-images.githubusercontent.com/44570474/255141170-0a6b076a-873a-4c38-a1d8-fd1c8309ad41.png)
> 
> ### Gmail
> ![image](https://user-images.githubusercontent.com/44570474/255142164-362bc7f7-6b99-4a9f-a016-e65e55c5206c.png)
> 
> Here I thought, hey maybe Tailwind is the problem so I wanted to recreate it using plain css, and here is the result.
> 
> ## Plain CSS
> ### Outlook
> ![image](https://user-images.githubusercontent.com/44570474/255145010-59f3399a-fec6-4ec0-aae4-ca7d2a83ea75.png)
> 
> ### Gmail
> ![image](https://user-images.githubusercontent.com/44570474/255144728-3701cff2-13b8-467b-bb00-f0b71ac9fc37.png)
> 
> Now I know that this is still in betaish but still this seems extremely weird... Or am I doing something wrong? Like am I not supposed to use flex or such things?

We have a bunch of users using in production, we use at Resend as well.
Would mind sharing the code?

**Reply by zanzlender** [reply]
I couldn't find the exact code from when I wrote this post, but here is another simple example.

```ts
import { Html } from "@react-email/html";
import { Button } from "@react-email/button";
import { Font } from "@react-email/font";
import { Preview } from "@react-email/preview";
import { Container } from "@react-email/Container";
import { Link } from "@react-email/link";
import { Heading } from "@react-email/heading";
import { Text } from "@react-email/text";
import { Head } from "@react-email/head";
import { Img } from "@react-email/img";
import { Body } from "@react-email/body";
import { Tailwind } from "@react-email/tailwind";
import * as React from "react";

export const NotionMagicLinkEmail = () => (
  <Html>
    <Head>
      <Font
        fontFamily="Comfortaa"
        fallbackFontFamily="Verdana"
        webFont={{
          url: "https://fonts.googleapis.com/css2?family=Comfortaa:wght@300;400;500;600;700&display=swap",
          format: "embedded-opentype",
        }}
        fontWeight={400}
        fontStyle="normal"
      />
    </Head>
    <Preview>Log in with this magic link</Preview>
    <Body style={{ backgroundColor: "#fffeb7" }}>
      <Tailwind>
        <Container className="mx-auto max-w-2xl px-4">
          <Img
            src="https://images.unsplash.com/photo-1682685797898-6d7587974771?ixlib=rb-4.0.3&ixid=M3wxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1170&q=80"
            width={300}
            className="mx-auto mb-20"
          />

          <Heading className="text-orange-600 font-bold text-center text-3xl mb-10">Login</Heading>

          <Text className="text-slate-700 mb-4">
            If you didn&apos;t try to login, you can safely ignore this email.
          </Text>

          <Text className="text-slate-700 mb-4 text-center">
            Klikni na gumb ispod i odvesti će te do stranice gdje možeš kreirati novu lozinku :)
          </Text>

          <Text className="text-slate-700 mb-4">Or, copy and paste this temporary login code:</Text>

          <Container className="mt-10 px-10 mx-auto py-6 border-2 border-orange-600 border-dashed flex justify-center gap-2 items-center w-fit">
            <Text className="inline font-semibold">Code:</Text>
            <Text className="inline font-bold">123456</Text>
          </Container>
        </Container>
      </Tailwind>
    </Body>
  </Html>
);

export default NotionMagicLinkEmail;

```

This time, Gmail looks as it should

![image](https://github.com/resendlabs/react-email/assets/44570474/d9ad943a-0943-4542-af0a-82bfb5c12d5b)

However, outlook does not.

![image](https://github.com/resendlabs/react-email/assets/44570474/c58c5241-9ff7-4a84-a95b-d50c475ed986)

And on both of those the font isn't loaded properly.

**Reply by zanzlender** [reply]
Also, for some reason if you set body background color with Tailwind then it doesn't work, but with styles it does...

**Reply by zanzlender** [reply]
Here is the vanilla version... Same results.

```ts
import { Html } from "@react-email/html";
import { Button } from "@react-email/button";
import { Font } from "@react-email/font";
import { Preview } from "@react-email/preview";
import { Container } from "@react-email/Container";
import { Link } from "@react-email/link";
import { Heading } from "@react-email/heading";
import { Text } from "@react-email/text";
import { Head } from "@react-email/head";
import { Img } from "@react-email/img";
import { Body } from "@react-email/body";
import { Tailwind } from "@react-email/tailwind";
import * as React from "react";

export const VanillaCSSEmailTest = () => (
  <Html>
    <Head>
      <Font
        fontFamily="Comfortaa"
        fallbackFontFamily="Verdana"
        webFont={{
          url: "https://fonts.googleapis.com/css2?family=Comfortaa:wght@300;400;500;600;700&display=swap",
          format: "embedded-opentype",
        }}
        fontWeight={400}
        fontStyle="normal"
      />
    </Head>
    <Preview>Log in with this magic link</Preview>
    <Body style={{ backgroundColor: "#fffeb7" }}>
      <Tailwind>
        <Container
          style={{
            marginLeft: "auto",
            marginRight: "auto",
            maxWidth: 672,
            paddingLeft: 16,
            paddingRight: 16,
          }}
        >
          <Img
            src="https://images.unsplash.com/photo-1682685797898-6d7587974771?ixlib=rb-4.0.3&ixid=M3wxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1170&q=80"
            width={300}
            style={{
              marginLeft: "auto",
              marginRight: "auto",
              marginBottom: 80,
            }}
          />

          <Heading
            style={{
              color: "#e47309",
              fontWeight: "bolder",
              textAlign: "center",
              fontSize: 36,
              marginBottom: 40,
            }}
          >
            Login
          </Heading>

          <Text
            style={{
              color: "#333333",
              marginBottom: 16,
            }}
          >
            If you didn&apos;t try to login, you can safely ignore this email.
          </Text>

          <Text
            style={{
              color: "#333333",
              marginBottom: 16,
              textAlign: "center",
            }}
          >
            Klikni na gumb ispod i odvesti će te do stranice gdje možeš kreirati novu lozinku :)
          </Text>

          <Text
            style={{
              color: "#333333",
              marginBottom: 16,
            }}
          >
            Or, copy and paste this temporary login code:
          </Text>

          <Container
            style={{
              marginTop: 40,
              padding: "24px 40px",
              marginLeft: "auto",
              marginRight: "auto",
              border: "2px dashed #e47309",
              display: "flex",
              gap: 2,
              justifyContent: "center",
              alignItems: "center",
              width: "fit-content",
            }}
          >
            <Text
              style={{
                display: "inline",
                fontWeight: "bold",
              }}
            >
              Code:
            </Text>
            <Text
              style={{
                display: "inline",
                fontWeight: "bold",
              }}
            >
              123456
            </Text>
          </Container>
        </Container>
      </Tailwind>
    </Body>
  </Html>
);

export default VanillaCSSEmailTest;

```

**Reply by FreiBj** [reply]
Did you find this out? I'm struggling with outlook as well...

**Reply by zanzlender** [reply]
Sadly no... I had no time to experiment too much with this so I just bought some email templates in the end :/

However I did notice some things... You need to be careful what and how you use it... For example, you shouldn't use flex, be careful how you center etc. It should work if you have a simpler design, ideally aligned left. Then it looks ok for all providers.

**Reply by jinsley8** [reply]
https://www.emailonacid.com/blog/article/email-development/12_things_you_must_know_when_developing_for_gmail_and_gmail_mobile_apps-2/#style

`4. Gmail only supports <style> in the <head>`
It may be better to put styles in a <style> block inside <head>

Some other issues with Outlook rendering:
https://www.emailonacid.com/blog/article/email-development/how-to-code-emails-for-outlook-2016/
